pkg/statistics/handle/bootstrap.go

```diff
@@ -53,10 +53,19 @@ type MaxTidRecord struct {
 	tid atomic.Int64
 }
 
-func (h *Handle) initStatsMeta4Chunk(is infoschema.InfoSchema, cache statstypes.StatsCache, iter *chunk.Iterator4Chunk) {
+func (h *Handle) initStatsMeta4Chunk(ctx context.Context, is infoschema.InfoSchema, cache statstypes.StatsCache, iter *chunk.Iterator4Chunk) {
 	var physicalID, maxPhysicalID int64
 	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
 		physicalID = row.GetInt64(1)
+
+		// Detect the context cancel signal, since it may take a long time for the loop.
+		// TODO: add context to TableInfoByID and remove this code block?
+		select {
+		case <-ctx.Done():
```

Comment:
if ctx.Err() != nil { return }

Reply:
done

Reply:
No I mean replace the whole select {...}.

Reply:
I think @lance6716 means we can use `if ctx.Err() != nil { return }` instead of line63-69.